src/PrestaShopBundle/Resources/config/services.yml

```diff
@@ -533,3 +533,102 @@ services:
           - ["setKernel", ["@kernel"]]
         tags:
           - {name: data_collector, template: "@PrestaShop/Admin/WebProfiler/config.html.twig", id: "config", priority: "-255"}
+
+
+    # Localization
+    prestashop.core.localization.cache.adapter:
+        class: Symfony\Component\Cache\Adapter\ArrayAdapter
+        public: true
+
+    prestashop.core.localization.cldr.cache.adapter:
+        class: Symfony\Component\Cache\Adapter\FilesystemAdapter
+        arguments:
+            - 'CLDR'
+            - 0
+            - '%kernel.cache_dir%/localization'
+        public: true
+
+    prestashop.core.localization.locale.repository:
+        class: PrestaShop\PrestaShop\Core\Localization\Locale\Repository
+        arguments:
+            - '@prestashop.core.localization.cldr.localeRepository'
+            - '@prestashop.core.localization.currency.repository'
+            - "@=service('translator').getLocale()"
+        public: true
+
+    prestashop.core.localization.currency.repository:
+        class: PrestaShop\PrestaShop\Core\Localization\Currency\Repository
+        arguments:
+            - '@prestashop.core.localization.currency.datasource'
+        public: true
+
+    prestashop.core.localization.cldr.localeRepository:
+        class: PrestaShop\PrestaShop\Core\Localization\CLDR\LocaleRepository
+        arguments:
+            - '@prestashop.core.localization.cldr.localeDataSource'
+        public: true
+
+    prestashop.core.localization.cldr.localeDataSource:
+        class: PrestaShop\PrestaShop\Core\Localization\CLDR\LocaleDataSource
+        arguments:
+            - '@prestashop.core.localization.cldr.middleware.topLayer'
+        public: true
```

Comment:
you don't need to set explicit "public" except if your class extends another "abstract" service

Reply:
ok.